### ISSUE TITLE
[BE] connect to VAMF FHIR sandbox

### ISSUE THREAD

**Issue by massrb** [department-of-veterans-affairs/va.gov-team#12778]
Tasks

- [ ] Find out how to connect to VAMF FHIR sandbox and can this be done outside of vets-api to make research easier
- [ ] read or write some records from the sandbox such as questionnaires, questionResponses or read patient data

Acceptance Criteria

- [ ] Tasks above completed

**Comment by abuckley3385** [comment]
@massrb  can you comment describe what the difference in this ticket compared to #12406 the descriptions and acceptance seem the same and IMO need to be combined.

**Comment by mchelen-gov** [comment]
@abuckley3385 it seems like this might a sub issue of https://github.com/department-of-veterans-affairs/va.gov-team/issues/12406
@massrb can you try using a tool like https://marketplace.visualstudio.com/items?itemName=humao.rest-client to manually query VAMF FHIR sandbox?

**Comment by massrb** [comment]
@mchelen-gov I have used this Ruby client on some fhir based sandboxes on the web for experimentation https://github.com/fhir-crucible/fhir_client. 

This slack discussion also mentions that we need access to the MAP PGD sandbox, so it's not currently accessible: https://dsva.slack.com/archives/C019EATUKN2/p1598538149005400

 I don't know if it's really open access to the fhir PGD because this document mentions an API gateway. If there is a gateway, that could imply that you have to use an API instead. https://wiki.mobilehealth.va.gov/display/PGDG/PGD+FHIR+Architecture+Overveiw

**Comment by mchelen-gov** [comment]
@massrb ah ok, yeah i see the "fhir client" mentioned in the https://github.com/department-of-veterans-affairs/va.gov-team/issues/12406#issue-677747029 tasks
wasn't sure if it would be easier to use a simple CURL or POSTman type query to start, or just go right for the FHIR client? 🤷‍♂️

**Comment by massrb** [comment]
@mchelen-gov I find it easy to work with Ruby based gems such as that fhir client. I had played with it on a public sandbox which I found it was very easy to connect to and pull down some fhir based patient records. I think it was this sandbox: https://www.logicahealth.org/solutions/fhir-sandbox/

 We don't have access to the VAMF (MAP) sandbox and are waiting on that.

**Comment by massrb** [comment]
The associated Jira ticket is ECCB-1617 and is marked as completed.

I got an email from Jakin Harper that was just a password. Currently I don't know how to use the PW in which context which I emailed Jakin about also mentioning that Stephen is out today. I did look through the wiki as well

**Comment by massrb** [comment]
Stephen is working on getting the keys to connect to the sandox:

>> With this [URL](https://vagovtest.mapsandbox.net/users/v2/), we should be able to start looking at the Auth flow based on this [documentation](https://coderepo.mobilehealth.va.gov/projects/IUMS/repos/user-service/browse/docs/externalJwtExchange.md).  Pretty sure we will want to use the /session endpoint. For that to be successful we will need the private/public keys set up.

```
Lynne said she would create the tickets for Staging and Prod once she gets the keys.  We need to determine who we ask for the keys from our side and have them send to her via encrypted email.  We will set up the Sandbox keys ourselves.
```

**Comment by massrb** [comment]
I met with Paul and Mark to discuss the priorities including this ticket. These are some notes I worked up for the meeting

https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/health-care/questionnaire/discovery/engineering/connecting%20to%20map%20(was%20vamf).md

**Comment by abuckley3385** [comment]
@massrb  Im not clear on the second section [View appointment details in questionnaire.](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/health-care/questionnaire/discovery/engineering/connecting%20to%20map%20(was%20vamf).md#view-appointment-details-in-questionnaire) The comments below this section are confusing viewing appointment details in our questionnaire with reading and writing from/to PGD. These (services and display of the data) are unrelated until the user submits the form and we need to send the Appt. details (RFV) thru MAP FHIR to PGD.   Can you let me know a bit more about what was discussed in relation to this section so I understand better?

**Comment by massrb** [comment]
@abuckley3385, yes I wanted to clarify that because the questionnaire was mentioned somewhere but what you are saying is what we agreed on

**Comment by massrb** [comment]
This is blocked until we have keys that allow access to the sandbox

**Comment by massrb** [comment]
This is basically a duplicate of #13156 and will be closed
https://app.zenhub.com/workspaces/vft-59c95ae5fda7577a9b3184f8/issues/department-of-veterans-affairs/va.gov-team/13156
